llvm/lib/Target/AMDGPU/SIPreEmitPeephole.cpp

```diff
@@ -326,15 +345,32 @@ bool SIPreEmitPeephole::mustRetainExeczBranch(
       if (TII->hasUnwantedEffectsWhenEXECEmpty(MI))
         return true;
 
-      // These instructions are potentially expensive even if EXEC = 0.
-      if (TII->isSMRD(MI) || TII->isVMEM(MI) || TII->isFLAT(MI) ||
-          TII->isDS(MI) || TII->isWaitcnt(MI.getOpcode()))
-        return true;
-
-      ++NumInstr;
-      if (NumInstr >= SkipThreshold)
+      if (TII->isWaitcnt(MI.getOpcode()))
         return true;
     }
+
+    if (!MinInstr)
+      MinInstr = Traces->getEnsemble(MachineTraceStrategy::TS_Local);
```

Comment:
Just initialize this in pass setup? Seems to be a constant?

Reply:
@arsenm what do you mean by "Just initialize in pass setup?"
This looks right to me. Unlike other analyses, for MTM, the getAnalysis returns the MTM pass object and the actual analysis/computation to be done exactly when & where they are needed.

Reply:
I mean at the start of runOnMachineFunction, just set MinInstr. There's not really a plus to deferring it until we see a branch